selectel/data_source_selectel_mks_kube_options.go

```diff
@@ -0,0 +1,243 @@
+package selectel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/selectel/mks-go/pkg/v1/kubeoptions"
+)
+
+type availableKubeOptionsSearchFilter struct {
+	kubeVersion string
+}
+
+func dataSourceFeatureGateTypeV1() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceFeatureGateTypeV1Read,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					ru1Region,
+					ru2Region,
+					ru3Region,
+					ru7Region,
+					ru8Region,
+					ru9Region,
+				}, false),
+			},
+			"filter": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"kube_version_minor": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"available_feature_gates": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"kube_version_minor": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"names": {
+							Type:     schema.TypeString,
```

Comment:
It is not useful to fetch all feature gate names as one joined string.
Let's change it to `schema.TypeSet` like you did it for "selectel_mks_cluster_v1" resource.